sklearn/gaussian_process/kernels.py

```diff
@@ -1320,9 +1320,9 @@ def __call__(self, X, Y=None, eval_gradient=False):
                 D = squareform(dists**2)[:, :, np.newaxis]
 
             if self.nu == 0.5:
-                K_gradient = K[..., np.newaxis] * D \
-                    / np.sqrt(D.sum(2))[:, :, np.newaxis]
-                K_gradient[~np.isfinite(K_gradient)] = 0
+                dist = np.ma.masked_array(np.sqrt(D.sum(2))[:, :, np.newaxis])
+                K_gradient = K[..., np.newaxis] * D / dist
+                K_gradient = K_gradient.filled(0)
```

Comment:
no, http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.ma.MaskedArray.filled.html

The notes say "The result is not a masked array"